### ISSUE TITLE
[BUG] Using feegrant, if a grantee doesn't have sufficient fees RLY errors out

### ISSUE THREAD

**Issue by dylanschultzie** [cosmos/relayer#1349]
After running the `feegrant` command:

```
rly chains configure feegrant basicallowance ...
```

If the grantee address doesn't have sufficient funds to cover the fees, the transaction will error out due to insufficient fees. It *should* be that the granter address is checked for sufficient funds. 

I've been getting around this by seeding initial amounts to the grantee addresses, but in theory this shouldn't be necessary.

**Comment by jtieri** [comment]
cc @KyleMoser

**Comment by KyleMoser** [comment]
Thanks, I will make sure this makes it into the update we're already rolling out.

**Comment by jonathanpberger** [comment]
Was this one fixed? @KyleMoser pls confirm

**Comment by KyleMoser** [comment]
@dylanschultzie Can you confirm this is fixed?

I ran the command `rly chains configure feegrant basicallowance` successfully with a new grantee address (unfunded) without issue. So I think this is fixed. I do want to make sure I'm not misunderstanding your bug report, though.

**Comment by KyleMoser** [comment]
@jonathanpberger Yes please close. I don't see any remaining issues but if I am mistaken, Dylan will have to give more info for me to reproduce.

**Comment by dylanschultzie** [comment]
@KyleMoser this should *not* be closed - apologies for not being clear enough...

```
rly chains configure feegrant basicallowance
```

This command runs successfully. Feegrant will properly be set.

However if you then try to relay (`rly start`) with said addresses not funded, the transaction fails due to having no fees. To be more precise, the transaction does not fail as it doesn't reach the failure state; `rly` provides an ERROR log stating there aren't sufficient fees to process the tx.

What I *think* is happening (imagine this as pseudocode):
```
if feegrant configured:
   check feegrantee address if sufficient funds for fees
```

what *should* be happening:
```
if feegrant configured:
   check **feegranter** address if sufficient funds for fees
```

**Comment by KyleMoser** [comment]
@dylanschultzie Can you please post the exact error log message? I still can't reproduce this issue. 

As you can see [here](https://github.com/cosmos/relayer/blob/8d9de865556fe315d1cbffff539840fe64d8223d/interchaintest/feegrant_test.go#L642) I changed the test case so it doesn't fund the grantees, and it still passes. I also reviewed the code we use to submit TXs and I don't see anything relevant. Clearly I'm missing something so maybe the exact error log will help me find the right spot.

**Comment by dylanschultzie** [comment]
Sure. It may be that it only happens on certain networks, but I'll review and come back with the specific error.

**Comment by KyleMoser** [comment]
@dylanschultzie Sorry for the delay but I think I've found some pretty good info indicating that this issue is now fixed. I think it is fixed [here](https://github.com/cosmos/relayer/commit/523748e91dd67145e4a1e4af6397e5814308cb28#diff-5209e4db1df3d26c385e6b9060[%E2%80%A6]2370a17c0063c6a400c1465ed637b8f2), where the gas calculation was changed.

- I backported my test case `TestRelayerFeeGrant` to commit 7ea0c479ec5d7dfa5bded2a087fe6ce9b989b610 (Nov 20th 2023). For that commit, the test case fails without funded grantees.
- For the latest commit, running TestRelayerFeeGrant succeeds regardless of whether the grantees are funded or not. 

In total, I think this confirms that the bug has been fixed.
